### PR TITLE
Update sc-view.js to render web components

### PR DESCRIPTION
**Pull request by itsMattShull**
This change allows web components to be rendered using the router by taking the child nodes from newView and using innerHTML to copy the HTML from newView into currentView.  This appears to render the HTML which triggers the web component lifecycle callbacks.

### CODE REVIEW

**Comment by itsMattShull** [comment]
@pmgower helped in talking this through.

**Comment by surma** [comment]
Can you give me a gist (or something) that reproduces the problem with ShadowDOM? I still would like to avoid using `innerHTML` and I don’t see why lifecycle callbacks wouldn’t get triggered when using `DocumentFragments` and `appendChild()`.

**Comment by pmgower** [comment]
@surma when we were testing we could reproduce the error in Chrome if we went to a page that had a custom element and hard refreshed to make sure nothing was cached. Then when we would go to one of the routes via the menu @derekshull had created in his demo when the view was displayed it would not render the shadowdom for the custom elements. @derekshull is using the callback methods of the v0 spec. We tried changing it to use v1 but couldn't get it to work. 

We are definitely open to a better way of doing this but this was the way we got it to work after several hours of pair programming yesterday.

**Comment by itsMattShull** [comment]
I'll post the code and an example to work test with once I knock out a couple of things this morning.

**Comment by itsMattShull** [comment]
Here's the code for sc-view: https://github.com/derekshull/styleguide/blob/gh-pages/js/components.js#L116

And here's an example to test: https://derekshull.github.io/styleguide/

Go to the initial page, then open the side nav and go to the buttons page.  In Chrome the custom elements won't render the shadow dom.  In Safari and FF, they will render properly.  Maybe first child isn't working in Chrome like it does in other browsers?  Or maybe appendChild isn't working in Chrome like other browsers?

You'll notice on line 116 that I'm using `document.createElement('div')` instead of `DocumentFragment()`.   I explain why in this issue: https://github.com/GoogleChrome/ui-element-samples/issues/57

**Comment by surma** [comment]
So I took a look and learned a few interesting few things:

TL;DR: _For now_, you’ll probably have to use `innerHTML`.

So, the reason FF and Safari behave differently is because they don’t have native support for ShadowDOM or Custom Elements yet, so the polyfill starts up. The polyfill uses normal DOM to emulate ShadowDOM so the elements get carried over with `appendChild()`. Bear in mind that it is technically still not working properly, as only the DOM is  carried over, nothing of the semantics.

Chrome _does_ have ShadowDOM and CustomElements. The reason lies in `xhr.responseType = 'document'`. XHR creates a `DocumentFragment`. `DocumentFragment`s have a _separate_ element registry, so that document knows nothing about any custom elements registered in the main document, so their code does not get run. By the time the DOM elements get carried over, the element has already degraded to a generic `HTMLElement` and no constructor or code will be run, hence no `ShadowRoot` will be created.

I’m still investigating, but it seems CustomElementsv1 and ShadowDOMv1 might address these problems. 

However, I’m not going to merge this PR. Supercharged is about performance and I feel like using `innerHTML` is to big of a tradeoff for a currently unstable API, especially when the next iterations of these APIs might be fixing the problem. I hope you understand :)

**Comment by surma** [comment]
In case you are interested, [here](https://gist.github.com/surma/266194af91ecab829f8da00c38ff76b6)’s my test code 😄

**Comment by itsMattShull** [comment]
Absolutely I understand 😀 that all makes sense. It's very interesting! I appreciate you digging in and investigating.

**Comment by surma** [comment]
Follow up: It works with CustomElements v1 + ShadowDOM v1.

Try this in Canary: https://gist.github.com/surma/5d91bae01a132cbacecd518b7178e89f
